kernelci-api/doc: update test report description (#147)
kernelci/kernelci-api: doc/overview.md

performing some static analysis etc.  Each environment needs to have its own
API token set up locally to be able to submit the results.

### Email Report

The Email Report in its current status listens for events once a build and its revisions'

> tests are completed and renders an email message using a jinja template, in the rendered

Reviewer: ```suggestion
tests are completed and renders an email message using a jinja template. In the rendered
```

message you can see relevant information about a build or revision. When the message is ready

> the application opens a connection to an STMP server, and send the report to an user or a list of

Reviewer: ```suggestion
the application opens a connection to an STMP server and sends the report to a user or a list of
```

> the application opens a connection to an STMP server, and sends the report to a user or a list of

Reviewer: ```suggestion
the application opens a connection to an STMP server and sends the report to a user or a list of
```

Author: @JenySadadia updated.

recipients that are waiting for the report with the build information after executing tests and

> possible regressions. With the report it is possible to see the number of tests have failed or passed,

Reviewer: ```suggestion
possible regressions. With the report, it is possible to see the number of tests that have failed or passed,
```

as seen in the example below:

> ```mermaid

Reviewer: As it does not include any diagram, why do we need `mermaid` here?

Author: @JenySadadia we do not need, it was wrong ;) all updated.

### Email Generator

The Email Generator step listens for pub/sub events when tests are complete
to then generate a summary and send it via email.  The current implementation

> renders an email message based on the test report jinja template, opens a conenction

Reviewer: ```suggestion
renders an email message based on the test report jinja template, opens a connection
```

> to a STMP server and sends a report message to a user or a list of recipients once

Reviewer: ```suggestion
to an STMP server, and sends a report message to a user or a list of recipients once
```

a change happened in the test results. At this moment emails are sent in case of a

> test fail or pass. A complete production solution would send emails to various

Reviewer: ```suggestion
test fails or passes. A complete production solution would send emails to various
```

Author: in this case fail and pass are nouns, no? To use fails or passes we need to replace "in case of" with "in case" as well.

Reviewer: I think in both cases, we need `fails or passes`. We should have @gctucker's review for this.

Reviewer 2: To be honest I'm not sure I understand the meaning of the sentence. Do you mean that email reports are sent either when a test is passing or failing? Or do you mean that the reports contain information about tests that passed and failed?

Author: At this moment this sentence doesn't make sense anymore..The reports is not sent in case of a test fail or pass. :)

recipients depending on the tests and would wait for an event signalling that

> a group of tests has completed (or timed out).

Reviewer: ```suggestion
a group of tests has been completed (or timed out).
```

Author: I am not sure about this as well, but I'd like to mean that the test was completed or even time out, and to use been I will need to update to something like: "has been completed or has been timed out". 

But I don't know, I am not a native speaker of English, maybe @gctucker can give us a hand.
